### PR TITLE
even better loader

### PR DESCRIPTION
**Pull request by Tcll**
there may be a few kinks, but it should find anything anything with "freetype" in it's name in any of: (in order)
1: the system directories
2: extended custom directories
3: current working directory
4: fail-safe extension directory

### CODE REVIEW

File: freetype/raw.py [hunk 1]
```diff
@@ -21,20 +21,52 @@
...
-filename = ctypes.util.find_library('freetype')
 
-if filename is None:
-    if platform.system() == 'Windows':
-        # Check current working directory for dll as ctypes fails to do so
-        filename = os.path.join(os.path.realpath('.'), 'freetype.dll')
-    else:
-        filename = 'libfreetype.so.6'
+def win_find_library(name):
+        if name in ('c', 'm'):
```
Comment (QuLogic) [inline freetype/raw.py]: Why the extra indent?

File: freetype/raw.py [hunk 2]
```diff
@@ -21,20 +21,52 @@
...
+                
+            if len(fname):
+                fname = fname[0]
+                if os.path.isfile(fname):
+                    return fname
+            continue
+        return ''
+
+if platform.system() == 'Windows': # add CWD to the search path
+    CWD = os.path.realpath('.')
```
Comment (QuLogic) [inline freetype/raw.py]: `os.getcwd` doesn't work?

File: freetype/raw.py [hunk 3]
```diff
@@ -21,20 +21,52 @@
...
 try:
-    _lib = ctypes.CDLL(filename)
+    lastcwd = os.getcwd()
+    os.chdir(filepath) # this method succeeds LoadLibrary(os.path.join(filepath,filename)) fails.
+    _lib = ctypes.cdll.LoadLibrary(filename)
+    os.chdir(lastcwd)
 except (OSError, TypeError):
     _lib = None
-    raise RuntimeError('Freetype library not found')
+    raise RuntimeError('Freetype library dependences not installed')
```
Comment (QuLogic) [inline freetype/raw.py]: dependencies

**Comment by rougier** [comment]
So what's the status of this patch in the end ? Does it solve the windows problem ?
What is the usual way of dealing with DLL ? (the setup now appears quite complex)

**Comment by mdsitton** [comment]
Well the globing would be nice to have so the dll names could be a little less strict. However this patch should not be merged in its current state, to many issues for me atm.

**Comment by Tcll** [comment]
yea, for the current status, you can check (and optionally download) raw.py here:
https://copy.com/9rNzsPqNaBPuyuXH
ultimately a local path was decided against for a custom bin path.

and yes it's working perfectly ;)
screenshot: http://lh3.ggpht.com/-SmxoWaZNJS4/VX2eVqs00XI/AAAAAAAAJQk/6-DIWZucgds/s802-Ic42/Screenshot%252520-%25252006142015%252520-%25252011%25253A26%25253A47%252520AM.png
<<< the box is the unicode refresh character.

**Pull request closed without merging**